code/permuter/RandomPermuter.cpp

```diff
@@ -46,6 +46,9 @@ bool RandomPermuter::gen_one_state(vector<epoch_op>& res,
   uniform_int_distribution<unsigned int> permute_requests(1,
       GetEpochs()->at(num_epochs - 1).ops.size());
   unsigned int num_requests = permute_requests(rand);
+  if (GetEpochs()->at(num_epochs - 1).ops.size() == 0) {
```

Comment:
Could you please add a comment in the code about what this fixes? I know we discussed it, but for other people looking at the code later, it will not be obvious what this addition actually does.